File: MinjiJu/2022-07-18-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        
        vector<vector<int>> res;
        if(intervals.size()==0) return res;
        
        sort(intervals.begin(), intervals.end());
        res.push_back(intervals[0]);
        
        int j=0;
        for(int i=0; i<intervals.size(); i++){
            if(res[j][1] >= intervals[i][0]) res[j][1] = max(res[j][1], intervals[i][1]);
            else{
                res.push_back(intervals[i]);
                j++;
            }
        }
        return res;
    }
};

File: Chris/2022-07-18-MergeIntervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        
        
        #sort the intervals by the starting point
        intervals.sort(key = lambda i : i[0])
        
        res = []
        
        
        for start, end in intervals:
            
            if not res or res[-1][1] < start:
                res.append([start,end])
            else:
                res[-1][1] = max(end, res[-1][1])
        
        
        return res
        
            
        
        